File: game_server/ClientProtocol/Common/CrossTeamEnum/ENPCrossTeamJoinCond.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using ALBasicProtocolPack;
using UnityEngine;


namespace Common.CrossTeamEnum
{

/// <summary>
/// 队伍加入条件
/// </summary>
public enum ENPCrossTeamJoinCond {
	NONE, //0 ==== 
	[InspectorName("MIN_POWER - idx[1] - 最小战力")]
	MIN_POWER, //1 ==== 最小战力
	[InspectorName("MIN_MARS_POWER - idx[2] - 最小火星战力")]
	MIN_MARS_POWER, //2 ==== 最小火星战力
}

public class ENPCrossTeamJoinCondComparer : IEqualityComparer<ENPCrossTeamJoinCond>{
	public bool Equals(ENPCrossTeamJoinCond x, ENPCrossTeamJoinCond y) { return x == y; }
	public int GetHashCode(ENPCrossTeamJoinCond obj) { return obj.GetHashCode(); }
	public static int g_iEnumCount = 3;
}
}

File: game_server/ServerProtocol/src/Common/CrossTeamEnum/ENPCrossTeamJoinCond.java
package Common.CrossTeamEnum;

/*********
 * 队伍加入条件
 **/
public enum ENPCrossTeamJoinCond {
	NONE, //0 ==== 
	MIN_POWER, //1 ==== 最小战力
	MIN_MARS_POWER, //2 ==== 最小火星战力
	;
public static final ENPCrossTeamJoinCond[]  ENPCrossTeamJoinCond_Values = ENPCrossTeamJoinCond.values();
public static final int ENPCrossTeamJoinCond_Length = ENPCrossTeamJoinCond_Values.length;
public static ENPCrossTeamJoinCond ENPCrossTeamJoinCond_FromInt(int _ivalue) {
	if(_ivalue < 0 || _ivalue >= ENPCrossTeamJoinCond_Length){ return null; }
	return ENPCrossTeamJoinCond_Values[_ivalue];
}
}

